src/components/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,307 @@
+</template>
+
+<script lang="ts">
+import { Vue, Options, prop } from 'vue-class-component'
+
+class Props {
+  /**
+   * A reactive native JS DateTime Object (only the date portion is reactive)
+   * @model
+   */
+  modelValue = prop<Date>({ required: false })
+  /**
+   * A native JS DateTime Object used to set the default value for the component
+   */
+  value = prop<Date>({ required: false })
+}
+
+const Component = Options
+@Component({
+  watch: {
+    value_(val) {
+      this.$emit('update:modelValue', val)
+    },
+    modelValue(val) {
+      this.date = new Date(val)
+    },
+    value(val) {
+      this.date = new Date(val)
+    }
+  }
+})
+
+/**
+ * The DatePicker is meant to present a unified cross-browser date picking component, replacing native browser date pickers
+ * which vary drastically browser to browser.
+ *
+ * @displayName DatePicker
+ */
+export default class DatePicker extends Vue.with(Props) {
+  value_: Date = this.modelValue
+    ? new Date(this.modelValue)
+    : this.value
+    ? new Date(this.value)
+    : new Date()
+
+  date: Date = new Date(this.value_)
+  monthDirection: number = 0
+
+  get today(): Date {
+    return new Date()
+  }
+
+  get previousMonthDate(): Date {
+    return new Date(this.year, this.month - 1)
+  }
+
+  get previousMonth(): number {
+    return this.previousMonthDate.getMonth()
+  }
+
+  get previousMonthYear(): number {
+    return this.previousMonthDate.getFullYear()
+  }
+
+  get nextMonthDate(): Date {
+    return new Date(this.year, this.month + 1)
+  }
+
+  get nextMonth(): number {
+    return this.nextMonthDate.getMonth()
+  }
+
+  get nextMonthYear(): number {
+    return this.nextMonthDate.getFullYear()
+  }
+
+  get displayMonth(): string {
+    const months: string[] = [
+      'January',
+      'February',
+      'March',
+      'April',
+      'May',
+      'June',
+      'July',
+      'August',
+      'September',
+      'October',
+      'November',
+      'December'
+    ]
+    return months[this.month]
+  }
+
+  getDisplayDay(day: number): string {
+    const days: string[] = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa']
+    return days[day]
+  }
+
+  get month(): number {
+    return new Date(this.date).getMonth()
+  }
+
+  get year(): number {
+    return new Date(this.date).getFullYear()
+  }
+
+  get daysInMonth(): number {
+    return 32 - new Date(this.year, this.month, 32).getDate()
+  }
+
+  get daysInNextMonth(): number {
+    return 32 - new Date(this.year, this.nextMonth, 32).getDate()
+  }
+
+  get daysInPreviousMonth(): number {
+    return 32 - new Date(this.year, this.previousMonth, 32).getDate()
+  }
+
+  get showToday(): boolean {
+    return this.month !== this.today.getMonth()
+  }
+
+  resetDate(): void {
+    const month = this.today.getMonth()
+    const year = this.today.getFullYear()
+    this.monthDirection =
+      (month >= this.month && year >= this.year) ||
+      (month < this.month && year > this.year)
+        ? 1
+        : -1
```

Comment:
This logic is duplicated. Maybe a `setMonthDirection` method that accepts date and compares it and sets `monthDirection`.

Reply:
Or can `resetDate` just call `selectDate`?